Lib/unittest/suite.py

```diff
@@ -199,14 +207,25 @@ def _handleModuleFixture(self, test, result):
             try:
                 setUpModule()
             except Exception as e:
+                try:
+                    case.doModuleCleanups()
+                except Exception as exc:
+                    self._createClassOrModuleLevelException(result, exc,
+                                                            'setUpModule',
+                                                            currentModule)
                 if isinstance(result, _DebugResult):
                     raise
                 result._moduleSetUpFailed = True
-                errorName = 'setUpModule (%s)' % currentModule
-                self._addClassOrModuleLevelException(result, e, errorName)
+                self._createClassOrModuleLevelException(result, e,
+                                                        'setUpModule',
+                                                        currentModule)
             finally:
                 _call_if_exists(result, '_restoreStdout')
 
+    def _createClassOrModuleLevelException(self, result, exc, parent, name):
```

Comment:
Shouldn't names `parent` and `name` be swapped or changed to more meaning names? The third argument is the method name and the forth argument is the module or class name.